src/codegen/compile.ml

```diff
@@ -11605,8 +11654,18 @@ and compile_prim_invocation (env : E.t) ae p es at =
     compile_exp_vanilla env ae e ^^
     IC.trap_text env
 
-  | OtherPrim ("blobToArray" | "blobToArrayMut"), e ->
-    const_sr SR.Vanilla (Arr.ofBlob env)
+  | OtherPrim "principalOfBlob", e ->
```

Comment:
When a `nop` turns into an alloc (+ linear copy) we should better have a good explanation for the refined tagging scheme :->